packages/features/settings/TimezoneChangeDialog.tsx

```diff
@@ -4,109 +4,122 @@ import { useSession } from "next-auth/react";
-      type="creation"
-      onInteractOutside={() => onCancel([1, "day"], false) /* 1 day expire */}>
+    <>
+      <DialogHeader
+        title={t("update_timezone_question")}
+        subtitle={t("update_timezone_description", { formattedCurrentTz })}
+      />
       {/* todo: save this in db and auto-update when timezone changes (be able to disable??? if yes, /settings)
         <Checkbox description="Always update timezone" />
         */}
       <div className="mb-8" />
       <DialogFooter showDivider>
-        <DialogClose onClick={() => onCancel([3, "months"], true)} color="secondary">
+        <DialogClose onClick={() => hideDialogFor([3, "months"], t("we_wont_show_again"))} color="secondary">
           {t("dont_update")}
         </DialogClose>
         <DialogClose onClick={() => updateTimezone()} color="primary">
           {t("update_timezone")}
         </DialogClose>
       </DialogFooter>
-    </DialogContent>
+    </>
   );
 };
 
 export function useOpenTimezoneDialog() {
   const { data: user } = trpc.viewer.me.useQuery();
   const [showDialog, setShowDialog] = useState(false);
-  const browserTimezone = dayjs.tz.guess() || "Europe/London";
-  const { isLocaleReady } = useLocale();
   const { data: userSession, status } = useSession();
 
+  const browserTimezone = dayjs.tz.guess() || "Europe/London";
+
   useEffect(() => {
-    if (
-      !isLocaleReady ||
-      !user?.timeZone ||
-      status !== "authenticated" ||
-      userSession?.user?.impersonatedBy
-    ) {
+    if (!user?.timeZone || status !== "authenticated" || userSession?.user?.impersonatedBy) {
       return;
     }
-    const cookie = document.cookie
-      .split(";")
-      .find((cookie) => cookie.trim().startsWith("calcom-timezone-dialog"));
-    if (
-      !cookie &&
-      dayjs.tz(undefined, browserTimezone).utcOffset() !== dayjs.tz(undefined, user.timeZone).utcOffset()
-    ) {
+    if (dayjs.tz(undefined, browserTimezone).utcOffset() !== dayjs.tz(undefined, user.timeZone).utcOffset()) {
       setShowDialog(true);
     }
-  }, [user, isLocaleReady, status, browserTimezone, userSession?.user?.impersonatedBy]);
+  }, [user?.timeZone, status, userSession?.user?.impersonatedBy]);
 
   return { open: showDialog, setOpen: setShowDialog, browserTimezone };
 }
 
-export default function TimezoneChangeDialog() {
-  const { open, setOpen, browserTimezone } = useOpenTimezoneDialog();
+function TimezoneChangeDialogContainer() {
+  const { open, setOpen } = useOpenTimezoneDialog();
+  const { t } = useLocale();
+
+  const handleInteractOutside = () => {
+    hideDialogFor([1, "day"], t("we_wont_show_again"));
+  };
+
   return (
     <Dialog open={open} onOpenChange={setOpen}>
-      <TimezoneChangeDialogContent browserTimezone={browserTimezone} onAction={() => setOpen(false)} />
+      {/* Dismiss dialog for a day on outside interaction */}
+      <DialogContent type="creation" onInteractOutside={handleInteractOutside}>
```

Comment:
Use DialogContent separately from TimezoneChangeDialogContent. Using this type of wrapper we ensure that TimezoneChangeDialogContent is only rendered when the DialogContent isOpen. Before the underlying component was always rendered.